plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts

```diff
@@ -0,0 +1,376 @@
+   */
+  private _getServerPatch(
+    original: (options?: Hapi.ServerOptions) => Hapi.Server
+  ) {
+    const instrumentation = this;
+    return function server(this: Hapi.Server, opts?: Hapi.ServerOptions) {
+      const newServer: Hapi.Server = original.apply(this, [opts]);
+
+      shimmer.wrap(newServer, 'route', original => {
+        return instrumentation._getServerRoutePatch.bind(instrumentation)(
+          original
+        );
+      });
+
+      shimmer.wrap(newServer, 'ext', original => {
+        return instrumentation._getServerExtPatch.bind(instrumentation)(
+          original
+        );
+      });
+
+      // Casting as any is necessary here due to multiple overloads on the Hapi.Server.register
+      // function, which requires supporting a variety of different types of Plugin inputs
+      shimmer.wrap(
+        newServer,
+        'register',
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        instrumentation._getServerRegisterPatch.bind(instrumentation) as any
+      );
+      return newServer;
+    };
+  }
+
+  /**
+   * Patches the plugin register function used by the Hapi Server. This function
+   * goes through each plugin that is being registered and adds instrumentation
+   * via a call to the @function _wrapRegisterHandler function.
+   * @param {RegisterFunction<T>} original - the original register function which
+   * registers each plugin on the server
+   */
+  private _getServerRegisterPatch<T>(
+    original: RegisterFunction<T>
+  ): RegisterFunction<T> {
+    const instrumentation = this;
+    this._logger.debug('Patching Hapi.Server register function');
+    return async function register(
+      this: Hapi.Server,
+      pluginInput: HapiPluginInput<T>,
+      options?: Hapi.ServerRegisterOptions
+    ) {
+      if (Array.isArray(pluginInput)) {
+        for (let i = 0; i < pluginInput.length; i++) {
+          instrumentation._wrapRegisterHandler(pluginInput[i].plugin);
+        }
+      } else {
+        instrumentation._wrapRegisterHandler(pluginInput.plugin);
+      }
+      await original.call(this, pluginInput, options);
+    };
+  }
+
+  /**
+   * Patches the Server.ext function which adds extension methods to the specified
+   * point along the request lifecycle. This function accepts the full range of
+   * accepted input into the standard Hapi `server.ext` function. For each extension,
+   * it adds instrumentation to the handler via a call to the @function _wrapExtMethods
+   * function.
+   * @param original - the original ext function which adds the extension method to the server
+   * @param {string} [pluginName] - if present, represents the name of the plugin responsible
+   * for adding this server extension. Else, signifies that the extension was added directly
+   */
+  private _getServerExtPatch(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    original: (...args: any) => void,
```

Comment:
should by `unknown` instead of `any` so you'll be forced to check types

Reply:
Thanks for your help with this!